[SITE-4597] WP-CLI 2.12 (#9582)
pantheon-systems/documentation: source/releasenotes/2025-06-18-wpcli-2-12.md

---
title: WP-CLI v2.12.0 release now available on Pantheon
published_date: "2025-06-18"
categories: [wordpress]
---

> We're thrilled to announce the availability of WP-CLI v2.12.0 on the Pantheon platform. Sites will automatically update to this version over the next 48 hours.

Reviewer: ```suggestion
We have updated the version of WP-CLI on the Pantheon platform to v2.12.0. Sites will automatically update to this version over the next 48 hours.
```
```suggestion
We're thrilled to announce the availability of WP-CLI v2.12.0 on the Pantheon platform. Sites will automatically update to this version over the next 48 hours.
```

WP-CLI is easily accessible for any WordPress site on Pantheon through our CLI, [Terminus](/terminus), allowing you to manage your WordPress installations efficiently with powerful commands at your fingertips.

### Highlights

#### Post lists can now handle complex query flags
When using `post list`, you can now use JSON notation to provide complex query arguments to the `--tax_query`, `--meta_query` and `--post_date` fields.

#### Post meta can be forced to only return a single value
The `post meta get` command now has a `--single` flag defaulting to `true` which can be negated with `--no-single`. This flag tells WordPress whether to only return a single value or all existing values for a given key.

#### Exclude files on core checksum verification
When running a `core checksum` verification, you can exclude one or more files from the checksum verification with the new `--exclude=<files>` flag.

#### Respect requires and requires_php tags for plugins and themes

> The `plugin` and `theme` commands now understand and respect the `requires` and `requires_php` header tags when trying ot install or update extensions. A new state `unavailable` has been introduced to denote the updates that are newer that your current installation but for which your site does not fulfill the requirements.

Reviewer 2: ```suggestion
The `plugin` and `theme` commands now understand and respect the `requires` and `requires_php` header tags when trying to install or update extensions. A new state `unavailable` has been introduced to denote the updates that are newer that your current installation but for which your site does not fulfill the requirements.
```
